benchmarks/benchmarks/analysis/distances.py

```diff
@@ -11,120 +17,243 @@
         atoms in two atomgroups with no offsets
         to resids.
         """
         distances.dist(A=self.ag1,
                        B=self.ag2,
+                       box=self.box_dims,
                        offset=0)
 
-    def time_dist_offsets(self, num_atoms):
+    def time_dist_offsets(self, num_atoms, box_dims):
         """Benchmark calculation of distances between
         atoms in two atomgroups with offsets
         to resids.
         """
         distances.dist(A=self.ag1,
                        B=self.ag2,
+                       box=self.box_dims,
                        offset=20)
 
-    def time_between(self, num_atoms):
-        """Benchmark determination of subgroup
-        of atomgroup that is within a specific
-        distance of two other atomgroups.
+
+class ContactsBench(object):
+    """Benchmarks for the MDAnalysis.analysis.distances.contact_matrix
+    function in both default and sparse settings.
+    """
+    u = MDAnalysis.Universe(GRO)
+    tri_clinic_dimensions = u.dimensions
+
+    params = ([10, 100, 1000], [None, 'orthogonal', 'triclinic'])
+    param_names = ['num_atoms', 'pbc_type']
+
+    def setup(self, num_atoms, pbc_type):
+
+        if pbc_type is None:
+            self.u = MDAnalysis.Universe(GRO)
+            self.box_dims = None
+
+        elif pbc_type == 'orthogonal':
+            # Some strange math/type issues. Essentially we create the smallest
+            # cubic box which encloses the triclinic unit cell.
+            self.u = MDAnalysis.Universe(GRO)
+            box_shape = self.u.dimensions
+            basis_vectors = mdamath.triclinic_vectors(box_shape)
+
+            # This will calculate the furthest point from the origin on the
+            # triclinic unit cell.
+            furthest_point = np.sum(basis_vectors, axis=0)
+
+            # Get full size of cube and deal with certain distance functions
+            # requiring float32.
+            orthogonal_box_size = np.float32(np.max(furthest_point))
+            angle = np.float32(90.0)
+
+            # make dummy cubic box, must be np.array, some functions do
+            # not accept list
+            self.box_dims = np.array([orthogonal_box_size,
+                                      orthogonal_box_size,
+                                      orthogonal_box_size,
+                                      angle, angle, angle])
+
+        elif pbc_type == 'triclinic':
+            self.u = MDAnalysis.Universe(GRO)
+            print(self.u.dimensions)
+            self.box_dims = self.u.dimensions
+        else:
+            raise Exception("Specify only 'orthogonal', 'triclinic', or None for the box type")
+
+        self.ag1 = self.u.atoms[:num_atoms]
+        self.ag2 = self.u.atoms[num_atoms: 2 * num_atoms]
+        self.ag3 = self.u.atoms[-num_atoms:]
+
+        np.random.seed(17809)
+        self.coords_1 = np.random.random_sample((num_atoms, 3)).astype(np.float32)
+        np.random.seed(9008716)
```

Comment:
Since we can use fairly new NumPy to run benchmarks, we may want to think about using the "newer"/[recommended mode](https://numpy.org/doc/stable/reference/random/new-or-different.html#new-or-different) of random number handling for new benchmarks eventually.

That said, I do find the "old way" far easier to understand, so probably fine for now.

Reply:
Happy to open a new issue if you like. I assume this will be an issue through a lot of the benchmarks module.